backends/bmv2/jsonconverter.cpp

```diff
@@ -803,13 +805,34 @@ JsonConverter::JsonConverter(const CompilerOptions& options) :
         headerParameter(nullptr), userMetadataParameter(nullptr), stdMetadataParameter(nullptr)
 {}
 
+void addSrcInfoData(Util::JsonObject* result,
```

Comment:
Umm, the design with visitors and mix-ins is there :) Unfortunately the implementation of the JSON converter is not conforming to the design. 

Or you could work together with @hanw and refactor it twice as fast! :)